preCICE support program (#112)
precice/precice.github.io: pages/community/community-support-precice.md

---

> title: You support preCICE - preCICE supports you

Reviewer: ```suggestion
title: You support preCICE – preCICE supports you
```

permalink: community-support-precice.html

> keywords:

Reviewer: ```suggestion
keywords: support, industry, contracting, service level agreement, price, on-site meeting
```

> summary:

Reviewer: TODO

toc: true
editme: false
---

## Why is support for preCICE necessary?

> Unfortunately the majority of open source software is abandoned at some point, because the developers lack the resources to maintain and develop the software. Using open source software in a professional context can therefore be considered risky, because the long-term sustainability is uncertain.  In order to ensure a sustainable future for preCICE, we are taking steps  to make sure preCICE remains the leading, state-of-the-art coupling solution. By supporting preCICE as an academic or industry user, you contribute to the long-term development of preCICE and will receive dedicated support from the preCICE development team.

Reviewer: ```suggestion
Unfortunately the majority of open source software is abandoned at some point, because the developers lack the resources to maintain and develop the software. Using open source software in a professional context can therefore be considered risky, because the long-term sustainability is uncertain.  In order to ensure a sustainable future for preCICE, we are taking steps  to make sure preCICE remains the leading, state-of-the-art coupling solution. By supporting preCICE as an academic or industry user, you contribute to the long-term development of preCICE and receive dedicated support from the preCICE development team.
```

## Funding and support options

There are three different ways to fund and support the development of preCICE.

### 1. Research project partner

> Include preCICE as a project partner in your research project proposal or team up with the preCICE developers to write a proposal. This is the best option when you want to benefit from the research and scientific expertise of the preCICE team, and increase the chances that your project is funded. The developers of preCICE are experts in state-of-the-art numerical methods, high-performance computing, and software engineering for multi-X coupled problems. Reviewers appreciate the highly recognized expertise of preCICE, demonstrated by our fast growing [number of citations](https://scholar.google.com/scholar?hl=en&cites=5053469347483527186), and [vibrant community](community-projects.html).

Reviewer: ```suggestion
Include preCICE as a project partner in your research project proposal or team up with the preCICE developers to write a proposal. This is the best option when you want to benefit from the research and scientific expertise of the preCICE team, and increase the chances that your project is funded. The developers of preCICE are experts in numerical methods, high-performance computing, and software engineering for multi-X coupled problems. Reviewers appreciate the highly recognized expertise of preCICE, demonstrated by our fast growing [number of citations](https://scholar.google.com/scholar?hl=en&cites=5053469347483527186), and [vibrant community](community-projects.html).
```

Directly [contact Jun. Prof. Benjamin Uekermann](https://www.ipvs.uni-stuttgart.de/departments/us3/) to discuss a collaboration.

> ### 2. Support license

Reviewer: Harald gave feedback that the word "license" might make things more complicated in an industrial environment as it might mean that the billing needs to go through the central software purchasing department. The obvious alternative would be "support contract". On the other hand, I think that the wording "license" could make it easier in an academic environment. Other opinions?

Author: That is a good point. Generally speaking everything a company above a certain size buys needs to go through purchasing as the default process. Also, very generally speaking, there will be exceptions and we should aim for these, but be prepared for the default.

What preCICE support offers is a service, however (independent of what the service is called), and it is hard to predict who the actual stakeholders in the purchasing process will be. I would keep license for now, and adapt when we know more.

#### Direct cost in a research project

As an academic user you can declare the support license, like any software license, as a direct cost/consumable in your research proposal (see e.g. section 2.1 in [basic module DFG guidelines](https://www.dfg.de/formulare/52_01/52_01_en.pdf)). It is also recommended to include the support license as a risk mitigation strategy to highlight that you, as a project lead, have the support for a successful implementation. In addition, as part of the support license, you can leverage the preCICE website and community for your communication/dissemination activities.

[Speak to Jun. Prof. Benjamin Uekermann](https://www.ipvs.uni-stuttgart.de/departments/us3/) to learn more.

#### Benefits

|   | Basic support license <br>(12 months, billed anually)  | Extended support license <br>(12 months, billed anually)  |
| - | - | -|
| **discounts for preCICE Workshop** | up to 3 tickets | up to 3 tickets |
| **priority support**   | via Discourse, up to 3 badges | via Discourse, up to 3 badges  |
| **visibility/dissemination**   | listed as supporter, project featured on website | listed as supporter, project featured on website  |

> | **private support**   |  | via email or Zoom |

Reviewer: ```suggestion
| **private support**   |  | via Discourse or Zoom |
```
There are also private communications on Discourse. I guess we would prefer these over email.

Reviewer 2: I would also change "Zoom" to something like "online meetings" / "videoconference".

We should also try the private communication features of Discourse, first.

> | **on-site support**    |  | up to 2 days of training or support |

Reviewer: ```suggestion
| **on-site support**    |  | up to 2 days of training or support per year |
```

> | **service level agreement** | | upon request |

Reviewer 2: What is a "service level agreement"? An actual contract? If this is not provided for the standard level, what do they get instead?

Author: Yes, that is correct, a contract that clarifies exactly what the levels of support are, how support is provided, typically with metrics (not applicable here but think of something like uptime), when and how the metrics are reviewed, by who (reporting structures), and what happens if something goes wrong etc.
If you buy software/service for everyday use, you don't get this level of support. Which is one of the reasons 'business software' is an order of magnitude more expensive than 'consumer software' even though the features of the software are very similar/the same.
So back to our case the standard level gets support as specified in the table but no guarantee, that say, somebody responds within a certain time frame.

Reviewer 2: > So back to our case the standard level gets support as specified in the table but no guarantee, that say, somebody responds within a certain time frame.

Ah, I see. This was not completely clear to me, so maybe it needs some further clarification. From the table I expected that this is the same for both levels. I guess for companies this is clear, but if we want to differentiate it, we could clrify it.

Author: Yes, if you need an SLA from your providers you should know 😆. Let's hope we don't have to provide one for now.

| **price** | upon request (discount for academia) | upon request (discount for academia) |

To request a support license [contact Jun. Prof. Benjamin Uekermann](https://www.ipvs.uni-stuttgart.de/departments/us3/).

### 3. (Sub-) contracting

> Contracting is another option to benefit from the expertise of the preCICE developers. As an academic user, this would typically be as a sub-contractor for a specific work package or task in your research project, e.g. active support in the development of a new adapter.

Reviewer: ```suggestion
Contracting is another option to benefit from the expertise of the preCICE developers. This option is the right choice if you require a new feature in preCICE itself or any of the official adapters. Development of an adapter for your code, on the other hand, is typically not well-suited for sub-contracting as this task requires expertise in your code, not in preCICE.
```
I removed the "as an academic ..." since I do not directly see a difference between academia and industry here.
And developing adapters could be dangerous here.

> Please note that we can only offer a very limited amount of contracting and we strongly advise you to include preCICE as a project partner, or acquire a support license instead.

Reviewer 2: The part of what we can actually help with and the part that there should preferably be some alignment with research (potentially with a link to the Roadmap) is missing here. Is this intentional at the moment?
